apm-agent-plugins/apm-asynchttpclient-plugin/src/main/java/co/elastic/apm/agent/asynchttpclient/AbstractAsyncHttpClientInstrumentation.java

```diff
@@ -178,8 +178,15 @@ public AsyncHandlerOnCompletedInstrumentation() {
         }
 
         @Advice.OnMethodEnter(suppress = Throwable.class)
-        private static void onMethodEnter(@Advice.This AsyncHandler<?> asyncHandler) {
-            final Span span = handlerSpanMap.remove(asyncHandler);
+        private static void onMethodEnter(@Advice.This AsyncHandler<?> asyncHandler, @Advice.Local("span") Span span) {
+            span = handlerSpanMap.remove(asyncHandler);
+            if (span != null) {
+                span.activate();
+            }
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class)
+        private static void onMethodExit(@Advice.This AsyncHandler<?> asyncHandler, @Nullable @Advice.Local("span") Span span) {
             if (span != null) {
```

Comment:
Don't forget to deactivate it in all exit advices.

Reply:
What is the difference between `end` and `deactivate`?

Reply:
`end` sets the end timestamp and schedules the span to be reported. `deactivate` removes the span from the thread local which avoids the span from being leaked into different executions.